calyx/src/passes/compile_ref.rs

```diff
@@ -13,8 +13,34 @@ use std::rc::Rc;
 /// 2. Inline all the ports of the ref cells to the component signature
 /// 3. Remove all the ref cell mappings from the invoke statement
 /// 4. Inline all the mappings of ports to the invoke signature
+
+/// Map for storing added ports for each ref cell
+/// level of Hashmap represents:
+/// HashMap<-component name-, Hashmap<-cell name-, HashMap<-port name-, port>>>;
+type RefPortMap =
+    HashMap<ir::Id, HashMap<ir::Id, HashMap<ir::Id, RRC<ir::Port>>>>;
+
+trait GetPorts {
+    fn get_ports(&self, comp_name: &ir::Id) -> Option<Vec<RRC<ir::Port>>>;
+}
+
+impl GetPorts for RefPortMap {
+    fn get_ports(&self, comp_name: &ir::Id) -> Option<Vec<RRC<ir::Port>>> {
```

Comment:
If you need to define methods on it, turn the `type` into a `struct` with exactly one element. Rust is clever enough to get rid of simple structs like that. Also, please document the method